the_mines/process/fussballdaten/process_matchday.py

```diff
@@ -0,0 +1,141 @@
+                    "a", attrs={"title": re.compile("Spieldetails:*")}
+                )
+                matchdate = date_re.findall(match_details[0]["title"])
+                matchdate = matchdate[0]
+                match = re.sub("\(\d+.\)", "", match.get_text())
+                match = umlaut(match)
+
+                team_re = re.compile("([\D]+)")
+                matchup = team_re.findall(match)
+
+                while ":" in matchup:
+                    matchup.remove(":")
+
+                # add matchup to dict with score
+                for team in matchup:
+                    if len(score_list[0]) == 0:
+                        del score_list[0]
+                    matchup_score.append((team, score_list[0].pop(0)))
+
+                matchdays[matchdate].append(matchup_score)
+
+            except IndexError:
+                pass
+
+            # if the match has not yet been played
+            if len(match_details) == 0:
+                try:
+                    match_details = match.find_all(
+                        "a", attrs={"title": re.compile("Vorschau:*")}
+                    )
+                    team_re = re.compile("\)([^|]*)\(")
+                    no_digits = re.compile("\D+")
+
+                    matchdate = date_re.findall(match_details[0]["title"])
+                    matchdate = matchdate[0]
+
+                    match = umlaut(match.get_text())
+                    # (16.)Düsseldorf15:30Dortmund(2.)- w -17,50X5,2521,36
+
+                    match = team_re.findall(match)
+                    # ['Düsseldorf15:30Dortmund']
+
+                    match = no_digits.findall(match[0])
+                    # ['Düsseldorf', ':', 'Dortmund']
+
+                    while ":" in match:
+                        match.remove(":")
+                    # ['Düsseldorf', 'Dortmund']
+
+                    matchup = match
+
+                    # add matchup to dict without score
+                    for team in matchup:
+                        matchup_score.append((team, "tbd"))
+
+                    matchdays[matchdate].append(matchup_score)
+
+                except IndexError:
+                    pass
+
+            # if the match is live
+            if len(match_details) == 0:
+
+                matchdate = date.today()
+                matchdate = matchdate.strftime("%d.%m.%Y")
+
+                # match = re.sub('\(\d+.\)', '', match.get_text())
+                match = umlaut(match.get_text())
+
+                team_re = re.compile("\)([^|]*)\(")
+                match = team_re.findall(match)
+
+                match = no_digits.findall(match[0])
+                while ":" in match:
+                    match.remove(":")
+                matchup = match
+                for team in matchup:
+                    matchup_score.append((team, "LIVE"))
+
+                matchdays[matchdate].append(matchup_score)
+    # print(matchdays)
+    return matchdays
```

Comment:
This section definitely looks better than it did before.  I still think there is room to improve.  There are several cases where indices are used where I think we could implement decompositions. Overall really nice work though this is very good!